### PR TITLE
Update README, add CONTRIBUTING and CODE_OF_CONDUCT

### PR DESCRIPTION
**Pull request by tschaffter**
Describe how to use this repository.

### CODE REVIEW

**Comment by thomasyu888** [comment]
Looks great @tschaffter.  One a side note:

Although we are proficient with the use of GitHub and git flow, not necessarily everyone is.  Take for example, our synapseDocs site (which is powered through GitHub / GitHub pages), where people are reluctant to contribute due to the use of GitHub / jekyll.  Not everyone is familiar with GitHub and unfortunately, "Edit Wiki" on Synapse is sometimes more user friendly if you're just editing a typo or small changes. We have to account for the fact that not everyone is as technical especially when contributing to documentation.

I am a huge fan of this solution, but I think it is a short term solution to what needs to be better addressed in the new challenge platform.  That being said, I think this solution works if we want someone more technical to be the main manager of the docs, which means, any changes of the doc site will only be pushed by that person.  The downside is, small things like: "Typo on this page" will be a GitHub issue, and what takes 2 seconds for someone to fix online would take a lot longer.

**Review by thomasyu888 — APPROVED** [review]

**Comment by tschaffter** [comment]
> Although we are proficient with the use of GitHub and git flow, not necessarily everyone is. Take for example, our synapseDocs site (which is powered through GitHub / GitHub pages), where people are reluctant to contribute due to the use of GitHub / jekyll. Not everyone is familiar with GitHub and unfortunately, "Edit Wiki" on Synapse is sometimes more user friendly if you're just editing a typo or small changes. We have to account for the fact that not everyone is as technical especially when contributing to documentation.

@thomasyu888 I agree with your statement. I tried to make the description as easy as possible to follow but giving a crash course on git/GitHub is out of the scope here. If someone comes across this repository and understands its README, then he/she may decide to use it. Otherwise they will ignore it and continue editing pages using Synapse UI.

> I am a huge fan of this solution, but I think it is a short term solution to what needs to be better addressed in the new challenge platform.

I don't see this solution as a short term solution but rather as an alternative solution for GitHub users to edit challenge pages. I think that the challenge platform that we will develop should definitively provide the API endpoints needed to support this solution.

**Comment by tschaffter** [comment]
@thomasyu888 I added more files to this PR and would appreciate your feedback on them. I tried to write the code of conduct and CONTRIBUTING.md so that we can add them almost as they are to the other repositories of this organization. This involved keep them relatively generic with project-specific instructions limited to a small number of sections.

**Comment by thomasyu888** [comment]
> I don't see this solution as a short term solution but rather as an alternative solution for GitHub users to edit challenge pages. I think that the challenge platform that we will develop should definitively provide the API endpoints needed to support this solution.

Ah, interesting.  I was thinking that if we provided a robust enough solution, we wouldn't need to edit the files locally.  Now that I think about it more, I realize that it is quite convenient to be able to edit the files locally.  I think we should push for a more global markdown implementation.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    npm run test
+
+## Release Procedure
+
+Maintainers are required to follow the procedure below when creating a new
+release.
+
+- Always merge all new features into `staging` branch first.
+- Update CHANGELOG.md with release information.
```
Comment (thomasyu888) [inline CONTRIBUTING.md]: Will we have `CHANGELOG.md` files?

File: README.md [hunk 2]
```diff
@@ -1,37 +1,116 @@
...
+After reviewing the pages on the staging site, a release of this GitHub
+repository can be created to mirror the content of the staging site to the
+production site.
 
-```
-challengeutils mirror-wiki syn22277124 syn22277123
-```
+### Create a new Synapse website
+
+Login to Synapse.org and create two Synapse projects, for example named
```
Comment (thomasyu888) [inline README.md]:
```suggestion
Login to synapse.org and create two Synapse projects, for example named
```

File: README.md [hunk 3]
```diff
@@ -1,37 +1,116 @@
...
+
+- `{Project name}`
+- `{Project name} - staging`
+
+Navigate to the staging site and create the structure of the pages under the
```
Comment (thomasyu888) [inline README.md:49]: The problem with creating a repository from this particular template is that people have to edit all the `parentId` to their own wiki pages.  This function isn't robust enough to create the wiki pages by linking the parent locally.  

Maybe the workflow could be:

```
challengeutils create-challenge ...
challengeutils pull-wiki
# make edits
challengeutils push-wiki
```

File: README.md [hunk 4]
```diff
@@ -1,37 +1,116 @@
...
 
 <!-- Links -->
 
+[NLP Sandbox website]: https://www.synapse.org/#!Synapse:syn22277123
 [NLP Sandbox Staging website]: https://www.synapse.org/#!Synapse:syn22277124
-[wiki]: wiki
\ No newline at end of file
+[NLP Sandbox Team]: https://github.com/orgs/nlpsandbox/people
+[GitHub issue]: https://github.com/nlpsandbox/nlpsandbox-website-synapse/issues/new/choose
+[Apache License 2.0]: https://github.com/elixir-cloud-aai/foca/blob/dev/LICENSE
```
Comment (thomasyu888) [inline README.md]:
```suggestion
[Apache License 2.0]: https://github.com/nlpsandbox/nlpsandbox-website-synapse/blob/staging/LICENSE
```

**Review by thomasyu888 — COMMENTED** [review]
Overall looks good, just some comments.

File: CONTRIBUTING.md (hunk 1 above)
Comment (tschaffter) [reply CONTRIBUTING.md]: I removed this line. Changelogs will be available in the Releases description.

File: README.md (hunk 3 above)
Comment (tschaffter) [reply README.md:49]: Later in the document I explain that the content of wiki should be emptied and that `pull-wiki` should be run. Also when creating the template repository we talk about, we could initially have the wiki folder empty, until the day where we can use `push-wiki` for the initial population of a wiki.

File: README.md [hunk 5]
```diff
@@ -1,37 +1,116 @@
...
+## Overview
 
-### Fetch pages from Synapse
+This repository includes the pages of the [NLP Sandbox website] hosted on
+Synapse.org. This repository is used by the [NLP Sandbox Team] to track changes
+made to the website pages and continuously deploy (CD) them to Synapse.
 
-The commands below were used to grab the pages of the [NLP Sandbox Staging website]
-in markdown format and save them to the directory [wiki].
+Using this reposiotry instead of Synapse wiki editor offers the following
```
Comment (vpchung) [inline README.md:11]:
```suggestion
Using this repository instead of Synapse wiki editor offers the following
```

File: README.md [hunk 6]
```diff
@@ -1,37 +1,116 @@
...
-```
-cd wiki
-challengeutils pull-wiki syn22277124
-```
+- More than one user to edit a page at the same time.
+- The community can propose changes using Pull Requests (PR).
+- Robust review process.
+- The source code of the pages (markdown) is automatically scanned by a linter
+  as part of the CI/CD process to improve code quality.
+- Faster page edition as it becomes possible to edit multiple pages at once
```
Comment (vpchung) [inline README.md:19]:
```suggestion
- Faster page editing as it becomes possible to edit multiple pages at once
```

File: README.md [hunk 7]
```diff
@@ -1,37 +1,116 @@
...
-cd wiki
-challengeutils push-wiki syn22277124
-```
+- Production site (e.g. [NLP Sandbox website])
+- Staging site
 
-## Mirroring staging site to production
+The default branch of this repository, `staging`, is used to automatically
+deploy changes made to this branch to the staging site on Synapse. We recommend
+configuring repositories created from this template to request that commits
```
Comment (vpchung) [inline README.md:33]: This sentence is a bit wordy and can be confusing.  It might help to break it up into bullet points? e.g.

"We recommend configuring repositories created from this template to stick to the following workflow:
* create a fork from the `staging` branch and make all edits/commits in that fork
* submit a PR from the fork into `staging`
* request one or more reviewers to accept the changes
* once approved, merge the changes into `staging` (this will trigger the deployment to the staging site)"

**Review by vpchung — APPROVED** [review]

**Comment by vpchung** [comment]
Sorry for the late review 😅

File: README.md (hunk 7 above)
Comment (tschaffter) [reply README.md:33]: Thanks! I adopted this version.
